AirCasting/SDSync/MobileSessions/SDSyncAveragingService.swift

```diff
@@ -0,0 +1,77 @@
+// Created by Lunar on 29/09/2022.
+//
+
+import Foundation
+import CoreLocation
+
+protocol SDSyncAveragingService {
+    func averageMeasurementsWithReminder<T: AverageableMeasurement>(measurements: [T], startTime: Date, averagingWindow: AveragingWindow, averagingAction: (T, [T]) -> Void) -> [T]
+}
+
+protocol AverageableMeasurement {
+    var time: Date! { get set }
```

Comment:
Would be better to not use explicitly unwrapped optionals in a `protocol` definition. I know that it's probably caused by a naming conflict. Let's rename time here to `measurementTime` or `measuredAt` and make it a `Date`